src/merkle_tree.rs

```diff
@@ -69,13 +80,27 @@ impl MerkleTree {
         MerkleTree { leaves: tree }
     }
 
+    /// This function is a wrapper over
+    /// [`MerkleTree::generate_proof_internal()`] to make calling the
+    /// function more convenvient. This filters out the indexes used
+    /// for debugging.
+    pub fn generate_proof<T: AsRef<[u8]>>(&self, elem: &T) -> Option<Vec<HashedData>> {
```

Comment:
If I understand correctly, the function `generate_proof` needs to return a vector of hashes and also the index of the leaf that we want to prove. Maybe you can return a tuple.

Reply:
You're 100% right! I'll change it ASAP